content/raw/chat/mutual_contacts.md

```diff
@@ -0,0 +1,80 @@
+
+## Use Cases
+
+### Sending a Contact Request
+
+1. User selects a profile and asks application sends contact request
+2. User fills reason for contact request e.g "Say who you are / why you want to become a contact.."
+3. User sends contact request
+4. System notifies user `Contact Request Send`
+5. System opens A 1 on 1 chat with that user, it displays that a contact request to that user is pending. The user cannot send messages on the 1 on 1 while the request is pending.
+
+### Receiving Contact Request
+
+1. System notifies user of a contact request (note: see activity center spec)
+2. User checks pending contact requests
+3. system displays the username of the requester, time of request, and the motivation text for the request
+4. User accepts contact request
+4.1 alternative: User rejects contact request
+5. system makes contact request as accepted or rejected
+
+## Functional Requirements
+
+User A wants to chat with User B and so adds User B as a contact, User B receives a contact request with a message and can either accept or reject. User A can only chat with User B if User B accepts the contact request.
+
+### User A that sends contact request
+- User A CANNOT send messages to User B if User B has not accepted the contact request
+- User A MAY send a contact request to User B containing a message
+- The message for the contact request MUST NOT exceeed 280 characters
+- If User A has sent a contact request they MUST see that the contact request is pending and the message they have sent
+- If User A has sent a contact request they MUST be able to cancel the contact request
+
+### User B that receives contact request
+- A User CANNOT receive messages from users that are not in their contact list
+- A User MAY receive contact requests from users that are not in their contact list
+- A User CANNOT receive contact requests from users that are already in their contact list
+- If a User has pending contact requests the contact requests section MUST be displayed
+- If a User has no pending contact requests the contact requests section MUST NOT be displayed
+- If a User receives a contact request the counter of message requests MUST increase
+
+### Contact Requests Section
+
+- A pending contact request MUST display the username, date, and message of the request
+- A User MAY Decline ALL pending contact requests
+- A User MAY Accept ALL pending contact requests
+- The user MUST be able to accept a contact request directly
+- The user MUST be able to decline & block a contact request directly in this type of activity item
+- Clicking the context menu of a contact request item MUST show 2 options: "view profile" and "Decline and Block"
+- If a contact request item is rejected it MUST disapear from the UI
+- If a contact request item is accepted it MUST disapear from the UI
+- Clicking on this type of activity item MUST open that chat and jump to that message
+
+## Notes
+
```

Comment:
I don't know if it should be part of the spec, but what happens in the following scenario:
1. Alice and Bob are already contacts
2. Alice loses her phone, but has the seed phrase, and restores her account in a new mobile device

Since Alice has no way to sync her previous contacts, what will happen in this case. Bob already has Alice in his contact list, but Alice will not be able to contact Bob without sending a contact request. The rule in line 59 says that `A User CANNOT receive contact requests from users that are already in their contact list`

Reply:
@richard-ramos very good point, this scenario is the main reason why mutual contacts haven't gone live in Status Mobile yet.  The solution to this problem that the mobile team have been working on (and that is now very close to completion) is that every 8 hours (or the next time a Status client goes online if the gap since it was last online was more than 8 hours), every Status client will upload an encrypted backup of the user's state to Waku.

Then if the user looses their phone or their laptop, as long as they login to Status with the same seed phrase within 30 days of the last backup to waku happening, all their user state is automatically restored.  One of the beauties of this backup to waku service is that it happens entirely in the background with zero user intervention.

This same mechanism will also kind-of sync any two devices that the user is logged into with the same keys, however we should discourage users from doing this and instead encourage them to set up direct syncing between devices, because syncing multiple devices via 8 hourly backups to waku is a much worse UX with a higher probability of user state change conflicts causing what a user will perceive as unpredictable behaviour.

The initial MVP of this 'back up to waku' mechanism that should be ready soon will only back up mutual contacts to waku.  Then after this is live, we will expand the scope of this service to eventually include exactly the same set of user state data that we sync between two devices.

If you have more questions about how this will work, ping Cammellos and he can point you in the right direction.